packet/adaptationfield.go

```diff
@@ -0,0 +1,578 @@
+	}
+	if af[4] == 0 {
+		return gots.ErrAdaptationFieldZeroLength
+	}
+	return nil
+}
+
+// initAdaptationField initializes the adaptation field to have all false flags
+// it will also occupy the remainder of the packet.
+func initAdaptationField(p *Packet) {
+	af := AdaptationField(p[:])
+	af[4] = 183  // adaptation field will take up the rest of the packet by Default
+	af[5] = 0x00 // no flags are set by default
+	for i := 6; i < len(af); i++ {
+		af[i] = 0xFF
+	}
+}
+
+// parseAdaptationField parses the adaptation field that is present in a packet.
+// this function is only used by packet functions.
+func parseAdaptationField(p *Packet) AdaptationField {
+	return AdaptationField(p[:])
+}
+
+// returns if the adaptation field has a PCR, this does not check for errors.
+func (af AdaptationField) hasPCR() bool {
+	return af.getBit(5, 0x10)
+}
+
+// returns if the adaptation field has an OPCR, this does not check for errors.
+func (af AdaptationField) hasOPCR() bool {
+	return af.getBit(5, 0x08)
+}
+
+// returns the adaptation field's Splicing Point flag, this does not check for errors.
+func (af AdaptationField) hasSplicingPoint() bool {
+	return af.getBit(5, 0x04)
+}
+
+// returns if the adaptation field has a Transport Private Data, this does not check for errors.
+func (af AdaptationField) hasTransportPrivateData() bool {
+	return af.getBit(5, 0x02)
+}
+
+// returns if the adaptation field has an adaptation field extension, this does not check for errors.
+func (af AdaptationField) hasAdaptationFieldExtension() bool {
+	return af.getBit(5, 0x01)
+}
+
+// pcrLength returns the length of the PCR, if there is no PCR then its length is zero
+func (af AdaptationField) pcrLength() int {
+	if af.hasPCR() {
+		return 6
+	}
+	return 0
+}
+
+const pcrStart = 6 // start of the pcr with respect to the start of the packet
+
+// opcrLength returns the length of the OPCR, if there is no OPCR then its length is zero
+func (af AdaptationField) opcrLength() int {
+	if af.hasOPCR() {
+		return 6
+	}
+	return 0
+}
+
+// opcrStart returns the start index of where the OPCR field should
+// be with respect to the start of the packet.
+func (af AdaptationField) opcrStart() int {
+	return pcrStart + af.pcrLength()
+}
+
+// spliceCountdownLength returns the length of the splice countdown,
+// if there is no splice countdown then its length is zero.
+func (af AdaptationField) spliceCountdownLength() int {
+	if af.hasSplicingPoint() {
+		return 1
+	}
+	return 0
+}
+
+// spliceCountdownStart returns the start index of where the splice
+// countdown field should be with respect to the start of the packet.
+func (af AdaptationField) spliceCountdownStart() int {
+	return pcrStart + af.pcrLength() + af.opcrLength()
+}
+
+// transportPrivateDataLength returns the length of the transport private data,
+// if there is no transport private data then its length is zero.
+func (af AdaptationField) transportPrivateDataLength() int {
+	if af.hasTransportPrivateData() {
```

Comment:
Invert this condition to return early and reduce the mass of text to the right (Go idiom).